soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -120,6 +134,17 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 	nrf_regulators_dcdcen_vddh_set(NRF_REGULATORS, true);
 #endif
 
+#if defined(NRF_GPIO_FORWARDER_FOR_NRF5340_CPUAPP_ENABLED)
+	static const uint8_t forwarded_psels[] = {
+		DT_FOREACH_CHILD(DT_INST(0, nordic_nrf_gpio_forwarder), ALL_GPIOS_IN_NODE)
```

Comment:
I think there is no technical obstacle to allowing the user to create as many of these nodes as they want. Something like this:

```
#define ALL_GPIOS_IN_FORWARDER(node_id) DT_FOREACH_CHILD(node_id, ALL_GPIOS_IN_NODE)

static const uint8 forwarded_psels[] = {
	DT_FOREACH_STATUS_OKAY(nordic_nrf_gpio_forwarder, ALL_GPIOS_IN_FORWARDER)
};
```

That way, the user doesn't have to use `&gpio_fwd { .... };` if they don't want to, and can create their own forwarder nodes as well. That is a bit more flexible and doesn't seem to cost much extra implementation complexity.

Am I missing something?

Reply:
Done